test/shapirowilk.jl

```diff
@@ -0,0 +1,95 @@
+using HypothesisTests, LinearAlgebra, Test, Random
+using StableRNGs
+@testset "ShapiroWilk" begin
+    @testset "SWcoeffs" begin
+        @test HypothesisTests.SWCoeffs(3).A == [sqrt(2.0) / 2.0]
+        @test length(HypothesisTests.SWCoeffs(3)) == 3
+
+        S = HypothesisTests.SWCoeffs(10)
+        @test length(S) == 10
+        @test lastindex(S) == 10
+        @test length(S.A) == 5
+
+        @test S.A[4] == S[4] == -S[7]
+        @test S.A[2] == S[2] == -S[9]
+
+        S = HypothesisTests.SWCoeffs(11)
+        @test length(S) == 11
+        @test lastindex(S) == 11
+        @test length(S.A) == 5
+
+        @test S[6] == 0.0
+        @test S.A[5] == S[5] == -S[7]
+        @test S.A[3] == S[3] == -S[9]
+        @test S.A[1] == S[1] == -S[11]
+
+        #anti-symmetry
+        S = HypothesisTests.SWCoeffs(20)
+        @test all([S[i] == -S[end-i+1] for i in eachindex(S)])
+
+        # Values obtained by calling `_swilkfort` fortran subroutine directly.
+
+        SWc10 = HypothesisTests.SWCoeffs(10)
+        res = SWc10.A .- [0.573715, 0.32897, 0.214349, 0.122791, 0.0400887]
+        @test norm(res, 1) ≈ 0.0 atol = length(SWc10) * eps(Float32)
+
+        SWc20 = HypothesisTests.SWCoeffs(20)
+        res = SWc20.A .- [0.473371, 0.32174, 0.255663, 0.208297, 0.16864, 0.133584, 0.101474, 0.0712893, 0.0423232, 0.0140351]
+        @test norm(res, 1) ≈ 0.0 atol = length(SWc20) * eps(Float32)
+
+        rng = StableRNG(0x5bca7c69b794f8ce)
+        X = sort(rand(rng, Float64, 10))
+        # W, pval, _ = swilkfort(X)
+        W = 0.9088434774710951
+        # pval = 0.2731410626084226
+        @test HypothesisTests.swstat(X, SWc10) ≈ W atol = eps(Float32)
+    end
+
+    @testset "ShapiroWilk" begin
+        # syntactic tests
+        @test_throws ArgumentError ShapiroWilkTest([1, 2])
+        @test_throws ArgumentError ShapiroWilkTest([1, 2, 3], N1=4)
+        @test_throws DimensionMismatch ShapiroWilkTest([1, 2, 3], HypothesisTests.SWCoeffs(4))
+
+        t = ShapiroWilkTest([1, 2, 3])
+        @test t.W == 1.0 # W is at most one but numerics makes it nextfloat(1.0)
+        @test pvalue(t) == 1.0
+        @test sprint(show, t) isa String
```

Comment:
This is a bit weak. How about testing the full output?

Reply:
usually testing the whole output is formatting/numerics prone;
I've added a few tests for `occursin` though; Let me know if that's fine with you

Reply:
Being formatting-prone is precisely the goal. :-)

Numeric instability can indeed be annoying, but in the `occursin` tests you added values seem to be reliably integers?

Reply:
ok, we're testing the whole string right now